### PR TITLE
avoid failing entire suites when Float16Array is undefined

### PR DESCRIPTION
**Pull request by bakkot**
Followup to https://github.com/web-platform-tests/wpt/pull/45483 per https://github.com/web-platform-tests/wpt/pull/45483#issuecomment-2165766114 (cc @saschanaz).

https://github.com/web-platform-tests/wpt/pull/46278 and https://github.com/web-platform-tests/wpt/pull/45670 did part of this already, so I think these are the only remaining cases.

### CODE REVIEW

File: FileAPI/blob/Blob-constructor.any.js [hunk 1]
```diff
@@ -290,7 +290,9 @@ test_blob(function() {
     new Int16Array([0x4150, 0x5353]),
     new Uint32Array([0x53534150]),
     new Int32Array([0x53534150]),
-    new Float16Array([2.65625, 58.59375]),
+    typeof Float16Array === 'undefined'
```
Comment (inexorabletash) [inline FileAPI/blob/Blob-constructor.any.js:293]: Do we have any precedent for "TODO(#nnn)" comments so we remember to go back and update these?

**Review by inexorabletash — APPROVED** [review]
Thanks, looks good!

**Comment by saschanaz** [comment]
I would be happier if we can make those run a separate `test()` for each type, but I guess this works for now.

File: FileAPI/blob/Blob-constructor.any.js (hunk 1 above)
Comment (annevk) [reply FileAPI/blob/Blob-constructor.any.js:293]: Yeah, we should add a comment. I don't think we have a consistent pattern however.

But given that Float64Array is split out below, perhaps we can do that for Float16Array as well? That would solve it entirely. The only follow-up then would potentially be merging it all into a single test again at some point in the future.

**Review by twiss — APPROVED** [review]
LGTM for the WebCryptoAPI/ part.

File: FileAPI/blob/Blob-constructor.any.js (hunk 1 above)
Comment (bakkot) [reply FileAPI/blob/Blob-constructor.any.js]: Split it into its own test.

Comment (saschanaz) [reply FileAPI/blob/Blob-constructor.any.js]: >The only follow-up then would potentially be merging it all into a single test again at some point in the future.

Or completely split them into all separate tests, it's easier to notice from wpt.fyi that way.

**Review by annevk — APPROVED** [review]
Thanks!

**Pull request merged**